### PR TITLE
Lovepreet dev

### PR DESCRIPTION
**Pull request by Lovepreetcode4each**
1) Role and permission in sidebar
2)remove grade  basic salary master from sidebar 
3) Added role section in side  bar 
4) Restructured the create  grade page  according  to new requirement 
5) Restructued the basic_grade_salary_master_listing page according  to new requirement

### CODE REVIEW

File: app/Http/Controllers/RoleController.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    public function edit(Request $request)
+    {
+        $RolePermission=RolePermission::where(['role_id' => $request->id])->get();
+        $role  = Role::where(['id' => $request->id])->first();
+        return Response()->json(['role' =>$role, 'RolePermission' =>$RolePermission]);
+				$validate = $validator->valid();
+    }
+
+    public  function  update(REQUEST $request)
```
Comment (kumarisavita) [inline app/Http/Controllers/RoleController.php:58]: Please check this 'REQUEST', this should be in small case?

File: app/Http/Controllers/RoleController.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        if ($validator->fails())
+       {
+           return response()->json(['errors'=>$validator->errors()->all()]);
+       }
+       $validate = $validator->valid();
+
+       $RolePermission=RolePermission::where(['role_id' => $validate['role_id']])->get(); 
+       if (!empty($RolePermission))
+       {
+        RolePermission::where('role_id', $validate['role_id'])->delete();
```
Comment (kumarisavita) [inline app/Http/Controllers/RoleController.php:74]: code indentation should be there.

File: app/Http/Controllers/UsersController.php [hunk 3]
```diff
@@ -68,10 +68,20 @@ public function register(UserRequest $request)
     public function showListing()
     {
         // Retrieve all users
-        $users = User::with('role')->get();
-        $all_roles = Role::all();
-        //dump($users); dd();
-        return view('Users.listing', compact('users', 'all_roles'));
+        $role_id = auth()->user()->role_id;
+       
+        if($role_id == "1"){
```
Comment (kumarisavita) [inline app/Http/Controllers/UsersController.php:73]: please make constant for these values that are need to be fixed. Create a constant in .env file and then call here

File: app/Http/Middleware/RolePermissionMiddleware.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+use Closure;
+use Illuminate\Http\Request;
+use Symfony\Component\HttpFoundation\Response;
+use App\Models\Module;
+use App\Models\RolePermission;
+use App\Models\User;
+use Illuminate\Routing\Route;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Redirect;
+use DB;
```
Comment (kumarisavita) [inline app/Http/Middleware/RolePermissionMiddleware.php:14]: why we are using this here?

File: resources/views/Role/role.blade.php [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+                        @forelse($page->module as $val)
+                        <div class="form-check col">
+                            <label class="form-check-label" for="{{'listing_page_'.$val->id}}">
+                                {{$val->module_name}}</label>
+                            <input class="form-check-input" name="role_permissions[]" type="checkbox"
+                                id="{{'listing_page_'.$val->id}}" value="{{$val->id}}">
+                        </div>
+                        @empty
+                        @endforelse
+                        <!--<div class="form-check col-md-4">
```
Comment (kumarisavita) [inline resources/views/Role/role.blade.php:78]: remove this commented code if not required.

File: resources/views/Role/role.blade.php [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+            }
+        });
+    }
+
+    function deleteRole(id) {
+        $('#role_id').val(id);
+        $("#deleteModal").modal('show');
+    }
+
+    function deleteRole(id) {
```
Comment (kumarisavita) [inline resources/views/Role/role.blade.php:283]: this method is defined 2 times, please check this and remove the duplicate one

**Review by kumarisavita — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/RoleController.php (hunk 1 above)
Comment (Lovepreetcode4each) [reply app/Http/Controllers/RoleController.php:58]: done

File: app/Http/Controllers/RoleController.php (hunk 2 above)
Comment (Lovepreetcode4each) [reply app/Http/Controllers/RoleController.php:74]: done

File: app/Http/Middleware/RolePermissionMiddleware.php (hunk 4 above)
Comment (Lovepreetcode4each) [reply app/Http/Middleware/RolePermissionMiddleware.php:14]: Ye  sare use ch  ne .. page  ch redirect, auth use ho reha

File: resources/views/Role/role.blade.php (hunk 5 above)
Comment (Lovepreetcode4each) [reply resources/views/Role/role.blade.php:78]: done

File: resources/views/layout/sections/sidebar.blade.php [hunk 7]
```diff
@@ -10,23 +13,42 @@
...
 
-      <li class="nav-item">
-
+      <?php
+        use App\Http\Controllers\DashboardController;
+        $permissions = DashboardController::permissions();
+        foreach($permissions as  $val){ ?>
+        
+         <li>
+         <a class="nav-link collapsed" href="{{ route($val['module'][0]['route_name']) }}">
```
[the remark this answers is not in the document]
Comment (Lovepreetcode4each) [reply resources/views/layout/sections/sidebar.blade.php:22]: done

**Review by kumarisavita — APPROVED** [review]

**Pull request merged**